keps/sig-windows/0000-20181221-windows-group-managed-service-accounts-for-container-identity.md

```diff
@@ -88,119 +93,243 @@ To achieve the scale and speed required for containers, Windows uses a group man
+A developer has a Windows web app that they would like to deploy in a container with Kubernetes. For example, it may have a web tier that they want to scale out running ASP.Net hosted in the IIS web server. Backend data is stored in a Microsoft SQL database, which all of the web servers need to access behind a load balancer. An Active Directory Group Managed Service Account is used to avoid hardcoded passwords, and the web servers run with that credential today. The SQL Database has a user with read/write access to that GMSA so the web servers can access it. When they move the web tier into a container, they want to preserve the existing authentication and authorization models.
 
 When this app moves to production on containers, the team will need to coordinate to make sure the right GMSA is carried over to the container deployment.
 
-1. Active Directory domain administrator will:
+1. An Active Directory domain administrator will:
 
   - Join all Windows Kubernetes nodes to the Active Directory domain.
-  - Provision the GMSA and gives details to application admin.
+  - Provision the GMSA and gives details to Kubernetes cluster admin.
   - Assign access to a AD security group to control what machines can use this GMSA. The AD security group should include all authorized Kubernetes nodes.
 
-2. A Kubernetes admin will create a configmap (in the desired namespace) and populate it with configuration information associated with the GMSA:
-  - Run a Windows PowerShell script to verify the account exists, and capture enough data to uniquely identify the GMSA into a JSON file (credspec). This doesn't contain any passwords or crypto secrets.
+2. A Kubernetes cluster admin (or cluster setup tools or an operator) will install a cluster scoped CRD (of kind GMSACredSpec) whose instances will be used to store GMSA credential spec configuration:
 
-Example credspec.json
 ```
-{
-  "CmsPlugins": [
-    "ActiveDirectory"
-  ],
-  "DomainJoinConfig": {
-    "DnsName": "contoso.com",
-    "Guid": "244818ae-87ca-4fcd-92ec-e79e5252348a",
-    "DnsTreeName": "contoso.com",
-    "NetBiosName": "DEMO",
-    "Sid": "S-1-5-21-2126729477-2524075714-3094792973",
-    "MachineAccountName": "WebApplication1"
-  },
-  "ActiveDirectoryConfig": {
-    "GroupManagedServiceAccounts": [
-      {
-        "Name": "WebApplication1",
-        "Scope": "DEMO"
-      },
-      {
-        "Name": "WebApplication1",
-        "Scope": "contoso.com"
-      }
-    ]
-  }
-}
+apiVersion: apiextensions.k8s.io/v1beta1
+kind: CustomResourceDefinition
+metadata:
+  name: gmsacredspecs.auth.k8s.io
+spec:
+  group: auth.k8s.io
+  version: v1alpha1
+  names:
+    kind: GMSACredSpec
```

Comment:
based on the CRI and docker references, would this be better called "GMSACredentialSpec" or "CredentialSpec"?